### PR TITLE
Support the official tailwindcss plugins

### PR DESCRIPTION
**Pull request by stecky**
This change would allow specifying an official plugin in the tailwind.config.js file and on build, it would install the plugins via npm so that they get bundled by postcss

### CODE REVIEW

**Review by BamButz — CHANGES_REQUESTED** [review]
Hey, this change is an excellent addition to this project.
I noticed 2 things, if you could adjust them it would be great. :)

File: sources/BamButz.MSBuild.TailwindCSS/Targets/BamButz.MSBuild.TailwindCSS.targets [hunk 1]
```diff
@@ -3,11 +3,57 @@
...
         <CallTarget Targets="CheckIfNodeJSIsInstalled" />
+
+        <GetTailwindPluginsList ConfigFilePath="$(ProjectDir)/tailwind.config.js">
+            <Output ItemName="Plugins" TaskParameter="PluginsOut" />
+        </GetTailwindPluginsList>
+
         <CallTarget Targets="InstallTailwindCSS" />
 
-        <Exec Command="node $(MSBuildThisFileDirectory)../index.js &quot;%(TailwindCSS.FullPath)&quot; &quot;%(TailwindCSS.RootDir)%(TailwindCSS.Directory)%(TailwindCSS.Filename).min.css&quot; &quot;$(ProjectDir.TrimEnd('\'))&quot;"
+        <Exec Command="npm install \%(Plugins.Identity) --silent"
```
[the remark this answers is not in the document]
Comment (stecky) [reply sources/BamButz.MSBuild.TailwindCSS/Targets/BamButz.MSBuild.TailwindCSS.targets]: removed

File: sources/BamButz.MSBuild.TailwindCSS/Targets/BamButz.MSBuild.TailwindCSS.targets [hunk 2]
```diff
@@ -3,11 +3,57 @@
...
+            <Using Namespace="Microsoft.Build.Framework"/>
+            <Using Namespace="Microsoft.Build.Utilities"/>
+            <Code Type="Fragment" Language="cs">
+<![CDATA[
+if (!File.Exists(ConfigFilePath))
+    return true;
+
+var cfgFileStr = File.ReadAllText(ConfigFilePath);
+var cfgJson = cfgFileStr.Split(new string[]{"module.exports = "}, StringSplitOptions.None)[1];
+cfgJson = Regex.Replace(cfgJson, @"require\('(.*)'\)", "\"$1\"");
```
[the remark this answers is not in the document]
Comment (stecky) [reply sources/BamButz.MSBuild.TailwindCSS/Targets/BamButz.MSBuild.TailwindCSS.targets]: done

**Review by BamButz — APPROVED** [review]
Looks good to me.

I will merge this, but before I can publish a new version we have to implement a functionality to check if new plugins were added to the config.
We need also to check for newer versions of the NPM packages, but because this is a general problem, I will open a new issue for this.

**Pull request merged**